rskj-core/src/test/java/co/rsk/peg/LegacyErpFederationTest.java

```diff
@@ -54,16 +55,102 @@ class LegacyErpFederationTest {
+            federator0PublicKey, federator1PublicKey, federator2PublicKey,
+            federator3PublicKey, federator4PublicKey, federator5PublicKey,
+            federator6PublicKey, federator7PublicKey, federator8PublicKey
+        );
+
+        networkParameters = bridgeConstants.getBtcParams();
         emergencyKeys = bridgeConstants.getErpFedPubKeysList();
         activationDelayValue = bridgeConstants.getErpFedActivationDelay();
-
         activations = mock(ActivationConfig.ForBlock.class);
         when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
 
         federation = createDefaultLegacyErpFederation();
     }
 
+    private LegacyErpFederation createDefaultLegacyErpFederation() {
+        List<FederationMember> standardMembers = FederationTestUtils.getFederationMembersWithBtcKeys(standardKeys);
+        Instant creationTime = ZonedDateTime.parse("2017-06-10T02:30:00Z").toInstant();
+        long creationBlockNumber = 0L;
+
+        return new LegacyErpFederation(
+            standardMembers,
+            creationTime,
+            creationBlockNumber,
+            networkParameters,
+            emergencyKeys,
+            activationDelayValue,
+            activations
+        );
+    }
+
+    @Test
+    void createInvalidLegacyErpFederation_nullErpKeys() {
+        emergencyKeys = null;
+        assertThrows(FederationCreationException.class, this::createDefaultLegacyErpFederation);
+    }
+
+    @Test
+    void createInvalidLegacyErpFederation_emptyErpKeys() {
+        emergencyKeys = new ArrayList<>();
+        assertThrows(FederationCreationException.class, this::createDefaultLegacyErpFederation);
+    }
+
+    @Test
+    void createValidLegacyErpFederation_oneErpKey() {
+        emergencyKeys = Collections.singletonList(emergencyKeys.get(0));
+        assertDoesNotThrow(this::createDefaultLegacyErpFederation);
+    }
+    @Test
+    void createInvalidLegacyErpFederation_negativeCsvValue() {
+        activationDelayValue = -100L;
+        assertThrows(FederationCreationException.class, this::createDefaultLegacyErpFederation);
+    }
+
+    @Test
+    void createInvalidLegacyErpFederation_zeroCsvValue() {
+        activationDelayValue = 0L;
+        assertThrows(FederationCreationException.class, this::createDefaultLegacyErpFederation);
+    }
+
+    @Test
+    void createInvalidLegacyErpFederation_aboveMaxCsvValue() {
+        activationDelayValue = ErpFederationRedeemScriptParser.MAX_CSV_VALUE + 1;
+        assertThrows(FederationCreationException.class, this::createDefaultLegacyErpFederation);
+    }
+
+    @Test
+    void createValidLegacyErpFederation_exactMaxCsvValue() {
+        activationDelayValue = ErpFederationRedeemScriptParser.MAX_CSV_VALUE;
+        assertDoesNotThrow(this::createDefaultLegacyErpFederation);
+    }
+
+    @Test
+    void createInvalidLegacyErpFederation_aboveMaxScriptSigSize() {
+        BtcECKey federator9PublicKey = BtcECKey.fromPublicOnly(
```

Comment:
Let's have the default one using 9 memebers, so we test with the maximum allowed. And here add only one new member and show that it fails